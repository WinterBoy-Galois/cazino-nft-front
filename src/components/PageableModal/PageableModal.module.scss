@import '../../styles/shared';

.slide-right-enter {
  opacity: 0;
  transform: translateX(100%);
}
.slide-right-enter-active {
  opacity: 1;
  transform: translateX(0%);
  transition: opacity 300ms, transform 300ms;
}
.slide-right-exit {
  opacity: 1;
  // transform: translateX(0%);
}
.slide-right-exit-active {
  opacity: 0;
  // transform: translateX(-100%);
  transition: opacity 100ms, transform 100ms;
}

.slide-left-enter {
  opacity: 0;
  transform: translateX(-100%);
}
.slide-left-enter-active {
  opacity: 1;
  transform: translateX(0%);
  transition: opacity 300ms, transform 300ms;
}
.slide-left-exit {
  opacity: 1;
  // transform: translateX(0%);
}
.slide-left-exit-active {
  opacity: 0;
  // transform: translateX(100%);
  transition: opacity 100ms, transform 100ms;
}
