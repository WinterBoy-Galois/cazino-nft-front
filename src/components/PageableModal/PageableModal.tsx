import React, { ReactNode, useReducer, Reducer, useEffect, useRef, useState } from 'react';
import { pageableModalReducer, PageableModalState, PageableModalAction } from './lib/reducer';
import Modal from '../Modal';
import { CSSTransition, SwitchTransition } from 'react-transition-group';
import styles from './PageableModal.module.scss';
import Footer from './components/Footer';

interface IProps {
  show: boolean;
  pages: ReactNode[];
  title?: ReactNode | ReactNode[];
  onClose?: () => void;
  modalClassName?: string;
  activePage?: number;
}

const exitTimeout = 100;

const PageableModal: React.FC<IProps> = ({
  show,
  pages,
  title: initialTitle,
  onClose,
  modalClassName,
  activePage = 0,
}) => {
  const [title, setTitle] = useState<string>();
  const [state, dispatch] = useReducer<Reducer<PageableModalState, PageableModalAction>>(
    pageableModalReducer,
    { activePage: 0, pageCount: 0, direction: 'right' }
  );

  useEffect(() => {
    dispatch({ type: 'INIT_PAGES', payload: { pages, activePage } });
  }, [pages, show, activePage]);

  const mainRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    setTimeout(() => {
      if (mainRef.current) {
        mainRef.current.scrollTop = 0;
      }
    }, exitTimeout);
  }, [mainRef, state.activePage]);

  useEffect(() => {
    if (Array.isArray(initialTitle)) {
      setTitle(initialTitle[state.activePage] as string);
    } else {
      setTitle(initialTitle as string);
    }
  }, [initialTitle, state.activePage]);

  const classNames = {
    enter: styles[`slide-${state.direction}-enter`],
    enterActive: styles[`slide-${state.direction}-enter-active`],
    exit: styles[`slide-${state.direction}-exit`],
    exitActive: styles[`slide-${state.direction}-exit-active`],
  };

  return (
    <Modal
      show={show}
      title={title}
      mainRef={mainRef}
      onClose={onClose}
      footer={
        <Footer
          pageCount={state.pageCount}
          activePage={state.activePage}
          onPreviousPage={() => dispatch({ type: 'PREVIOUS_PAGE' })}
          onNextPage={() => dispatch({ type: 'NEXT_PAGE' })}
          onGoToPage={page => dispatch({ type: 'GO_TO_PAGE', payload: { page } })}
        />
      }
      modalClassName={modalClassName}
    >
      <SwitchTransition>
        <CSSTransition
          classNames={classNames}
          key={state.activePage}
          timeout={{ enter: 300, exit: exitTimeout }}
        >
          {pages[state.activePage]}
        </CSSTransition>
      </SwitchTransition>
    </Modal>
  );
};

export default PageableModal;
