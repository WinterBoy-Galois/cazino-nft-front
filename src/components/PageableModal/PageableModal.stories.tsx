import React from 'react';
import { storiesOf } from '@storybook/react';
import PageableModal from '.';

storiesOf('Components/PageableModal', module).add('default', () => (
  <PageableModal
    show={true}
    pages={[<div key={0}>Page 1</div>, <div key={1}>Page 2</div>, <div key={2}>Page 3</div>]}
  />
));
