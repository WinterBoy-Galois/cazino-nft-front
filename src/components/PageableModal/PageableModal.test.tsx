import React from 'react';
import { render } from '@testing-library/react';
import PageableModal from '.';

describe('PageableModal', () => {
  it('should match snapshot', () => {
    // Arrange

    // Act
    const container = render(<PageableModal />);

    // Assert
    expect(container).toMatchSnapshot();
  });
});
