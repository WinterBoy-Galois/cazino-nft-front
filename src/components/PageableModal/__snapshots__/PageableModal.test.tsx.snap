// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageableModal should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden;"
  >
    <div>
      <div
        class="container"
        style="height: 768px;"
      >
        <div
          aria-modal="true"
          class="modal scale--enter scale--enter-active"
          role="dialog"
        >
          <div
            class="modal__header"
          >
            <div
              class="modal__header__headline"
            >
              
            </div>
            <div
              class="modal__header__close"
            >
              <svg
                class="icon icon__close"
                viewBox="0 0 365.696 365.696"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
                />
              </svg>
            </div>
          </div>
          <div
            class="modal__main"
          >
            <div>
              test
            </div>
          </div>
          <div
            class="modal__footer"
          >
            <div
              class="container"
            >
              <div>
                <button
                  class="button"
                >
                  <svg
                    class="arrow arrow--left"
                    viewBox="0 0 12 26"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style>
                        .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                      </style>
                    </defs>
                    <g
                      data-name="Layer 2"
                      id="Layer_2"
                    >
                      <g
                        data-name="Layer 1"
                        id="Layer_1-2"
                      >
                        <g
                          data-name="Group 197"
                          id="Group_197"
                        >
                          <path
                            class="cls-1"
                            d="M1.5 24.5l9-11"
                            data-name="Line 23"
                            id="Line_23"
                          />
                          <path
                            class="cls-1"
                            d="M1.5 1.5l9 12"
                            data-name="Line 24"
                            id="Line_24"
                          />
                        </g>
                      </g>
                    </g>
                  </svg>
                </button>
              </div>
              <div
                class="dots"
              >
                <div
                  class="dot__wrapper dot__wrapper--active"
                >
                  <div
                    class="dot"
                  />
                </div>
              </div>
              <div>
                <button
                  class="button"
                >
                  <svg
                    class="arrow arrow--right"
                    viewBox="0 0 12 26"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <defs>
                      <style>
                        .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                      </style>
                    </defs>
                    <g
                      data-name="Layer 2"
                      id="Layer_2"
                    >
                      <g
                        data-name="Layer 1"
                        id="Layer_1-2"
                      >
                        <g
                          data-name="Group 197"
                          id="Group_197"
                        >
                          <path
                            class="cls-1"
                            d="M1.5 24.5l9-11"
                            data-name="Line 23"
                            id="Line_23"
                          />
                          <path
                            class="cls-1"
                            d="M1.5 1.5l9 12"
                            data-name="Line 24"
                            id="Line_24"
                          />
                        </g>
                      </g>
                    </g>
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
      style="height: 768px;"
    >
      <div
        aria-modal="true"
        class="modal scale--enter scale--enter-active"
        role="dialog"
      >
        <div
          class="modal__header"
        >
          <div
            class="modal__header__headline"
          >
            
          </div>
          <div
            class="modal__header__close"
          >
            <svg
              class="icon icon__close"
              viewBox="0 0 365.696 365.696"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M243.188 182.86L356.32 69.726c12.5-12.5 12.5-32.766 0-45.247L341.238 9.398c-12.504-12.503-32.77-12.503-45.25 0L182.86 122.528 69.727 9.374c-12.5-12.5-32.766-12.5-45.247 0L9.375 24.457c-12.5 12.504-12.5 32.77 0 45.25l113.152 113.152L9.398 295.99c-12.503 12.503-12.503 32.769 0 45.25L24.48 356.32c12.5 12.5 32.766 12.5 45.247 0l113.132-113.132L295.99 356.32c12.503 12.5 32.769 12.5 45.25 0l15.081-15.082c12.5-12.504 12.5-32.77 0-45.25zm0 0"
              />
            </svg>
          </div>
        </div>
        <div
          class="modal__main"
        >
          <div>
            test
          </div>
        </div>
        <div
          class="modal__footer"
        >
          <div
            class="container"
          >
            <div>
              <button
                class="button"
              >
                <svg
                  class="arrow arrow--left"
                  viewBox="0 0 12 26"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <style>
                      .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                    </style>
                  </defs>
                  <g
                    data-name="Layer 2"
                    id="Layer_2"
                  >
                    <g
                      data-name="Layer 1"
                      id="Layer_1-2"
                    >
                      <g
                        data-name="Group 197"
                        id="Group_197"
                      >
                        <path
                          class="cls-1"
                          d="M1.5 24.5l9-11"
                          data-name="Line 23"
                          id="Line_23"
                        />
                        <path
                          class="cls-1"
                          d="M1.5 1.5l9 12"
                          data-name="Line 24"
                          id="Line_24"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
              </button>
            </div>
            <div
              class="dots"
            >
              <div
                class="dot__wrapper dot__wrapper--active"
              >
                <div
                  class="dot"
                />
              </div>
            </div>
            <div>
              <button
                class="button"
              >
                <svg
                  class="arrow arrow--right"
                  viewBox="0 0 12 26"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <defs>
                    <style>
                      .cls-1{fill:none;stroke:#f4feff;stroke-linecap:round;stroke-width:3px}
                    </style>
                  </defs>
                  <g
                    data-name="Layer 2"
                    id="Layer_2"
                  >
                    <g
                      data-name="Layer 1"
                      id="Layer_1-2"
                    >
                      <g
                        data-name="Group 197"
                        id="Group_197"
                      >
                        <path
                          class="cls-1"
                          d="M1.5 24.5l9-11"
                          data-name="Line 23"
                          id="Line_23"
                        />
                        <path
                          class="cls-1"
                          d="M1.5 1.5l9 12"
                          data-name="Line 24"
                          id="Line_24"
                        />
                      </g>
                    </g>
                  </g>
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
